projects/types.nix

```diff
@@ -196,6 +196,53 @@ let
         };
       };
 
+    demo = types.submodule (
+      { name, ... }:
+      {
+        options = {
+          inherit (types'.example.getSubOptions { })
+            module
+            tests
+            description
+            links
+            ;
+          inherit (types')
+            problems
+            ;
+          kind = mkOption {
+            type = types.enum [
+              "vm"
+              "shell"
+            ];
+            default = name;
+          };
+        };
+      }
+    );
+
+    problems = lib.mkOption {
+      type = types.attrsOf types'.problem;
+      default = { };
+    };
+
+    problem = types.submodule {
```

Comment:
Same here, why not `problem.type = attrTag ...`? And then only define what we know so far: broken with reason.